oauth2/oauth2-resource-server/src/main/java/org/springframework/security/oauth2/server/resource/authentication/JwtAuthenticationConverter.java

```diff
@@ -16,57 +16,70 @@
 
 package org.springframework.security.oauth2.server.resource.authentication;
 
-import java.util.Arrays;
 import java.util.Collection;
-import java.util.Collections;
-import java.util.stream.Collectors;
 
 import org.springframework.core.convert.converter.Converter;
 import org.springframework.security.authentication.AbstractAuthenticationToken;
 import org.springframework.security.core.GrantedAuthority;
-import org.springframework.security.core.authority.SimpleGrantedAuthority;
 import org.springframework.security.oauth2.jwt.Jwt;
-import org.springframework.util.StringUtils;
+import org.springframework.util.Assert;
 
 /**
  * @author Rob Winch
  * @author Josh Cummings
  * @since 5.1
  */
 public class JwtAuthenticationConverter implements Converter<Jwt, AbstractAuthenticationToken> {
-	private static final String SCOPE_AUTHORITY_PREFIX = "SCOPE_";
+	private Converter<Jwt, Collection<GrantedAuthority>> authorityConverter = new JwtAuthorityConverter();
 
-	private static final Collection<String> WELL_KNOWN_SCOPE_ATTRIBUTE_NAMES =
-			Arrays.asList("scope", "scp");
+	/**
+	 * Create default instance using {@link JwtAuthorityConverter} as the authority converter
+	 */
+	public JwtAuthenticationConverter() {
+		super();
+	}
 
+	/**
+	 * Create an instance using a supplied {@link Converter Converter&lt;Jwt, Collection&lt;GrantedAuthority&gt;&gt;}
+	 *
+	 * @param authorityConverter The converter
+	 * @since 5.2
+	 */
+	public JwtAuthenticationConverter(Converter<Jwt, Collection<GrantedAuthority>> authorityConverter) {
```

Comment:
Please remove this constructor.

Reply:
See comments above on `ReactiveJwtAuthenticationConverter` on discussion about constructor vs setter. I think whatever decision we take there applies here too.